lang/java/reef-experimental/src/main/java/org/apache/reef/experimental/ParquetReader.java

```diff
@@ -0,0 +1,176 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.experimental;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.avro.AvroSchemaConverter;
+import org.apache.parquet.format.converter.ParquetMetadataConverter.MetadataFilter;
+import static org.apache.parquet.format.converter.ParquetMetadataConverter.NO_FILTER;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+
+/**
+ * A reader for Parquet files that can serialize data to local disk and return Avro schema and Avro reader.
+ * The intent is not to build a general parquet reader, but to consume data with table-like property.
+ */
+public final class ParquetReader {
+  /**
+   * Standard java logger.
+   */
+  private static final Logger LOG = Logger.getLogger(ParquetReader.class.getName());
+  
+  private Path parquetFilePath;
+
+  @NamedParameter(default_value="", doc="Path to parquet file", short_name="path")
+  class PathString implements Name<String> { }
+
+  @Inject
+  private ParquetReader(@Parameter(ParquetReader.PathString.class) final String path) throws IOException {
+    parquetFilePath = new Path(new File(path).getAbsolutePath());
+    final Schema schema = getAvroSchema();
+    if (schema.getType() != Schema.Type.RECORD) {
+      LOG.log(Level.SEVERE, "ParquetReader only support Avro record type that can be consumed as a table.");
```

Comment:
If this can't possibly work, then we should through the appropriate exception here.